Snowflake.Data.Tests/ConcatenatedStreamTest.cs

```diff
@@ -0,0 +1,159 @@
+        {
+            // Do nothing;
+        }
+
+        [Test]
+        public void TestCanRead()
+        {
+            Assert.IsTrue(concatStream.CanRead);
+        }
+
+        [Test]
+        public void TestCanSeek()
+        {
+            Assert.IsFalse(concatStream.CanSeek);
+        }
+
+        [Test]
+        public void TestCanWrite()
+        {
+            Assert.IsFalse(concatStream.CanWrite);
+        }
+
+        [Test]
+        public void TestFlush()
+        {
+            try
+            {
+                concatStream.Flush();
+            }
+            catch (Exception ex)
+            {
+                Assert.IsTrue(ex is NotImplementedException);
+            }
+        }
+
+        [Test]
+        public void TestLength()
+        {
+            try
+            {
+                long l = concatStream.Length;
+            }
+            catch (Exception ex)
+            {
+                Assert.IsTrue(ex is NotImplementedException);
+            }
+        }
+
+        [Test]
+        public void TestPosition()
+        {
+            try
+            {
+                concatStream.Position = 1;
+            }
+            catch (Exception ex)
+            {
+                Assert.IsTrue(ex is NotImplementedException);
+            }
+
+            try
+            {
+                long l = concatStream.Position;
+            }
+            catch (Exception ex)
+            {
+                Assert.IsTrue(ex is NotImplementedException);
+            }
+        }
+
+        [Test]
+        public void TestSeek()
+        {
+            try
+            {
+                concatStream.Seek(1, new SeekOrigin());
+            }
+            catch (Exception ex)
+            {
+                Assert.IsTrue(ex is NotImplementedException);
+            }
+        }
+
+        [Test]
+        public void TestSetLength()
+        {
+            try
+            {
+                concatStream.SetLength(1);
+            }
+            catch (Exception ex)
+            {
+                Assert.IsTrue(ex is NotImplementedException);
+            }
+        }
+
+        [Test]
+        public void TestWrite()
+        {
+            try
+            {
+                concatStream.Write(null, 0, 0);
+            }
+            catch (Exception ex)
+            {
+                Assert.IsTrue(ex is NotImplementedException);
+            }
+        }
+
+        [Test]
+        public void TestReadZeroByte()
+        {
+            byte[] buffer = new byte[0];
+            Assert.AreEqual(0, concatStream.Read(buffer, 0, 0)); // Read 0 byte
+        }
+
+        [Test]
+        public void TestReadBytes()
+        {
+            byte[] buffer = new byte[3];
+            Assert.AreEqual(1, concatStream.Read(buffer, 0, 1)); // Read 1 byte
+            Assert.AreEqual(2, concatStream.Read(buffer, 0, 2)); // Read 2 bytes
+            Assert.AreEqual(3, concatStream.Read(buffer, 0, 3)); // Read 3 bytes
```

Comment:
do we have a test where we try to read 4 bytes?

Reply:
I'll add a test for that